supplementary_content/views.py

```diff
@@ -46,19 +47,36 @@ class Meta:
 class SupplementaryContentView(generics.ListAPIView):
     serializer_class = CategorySerializer
 
-    section_list = []
-    title = None
-    part = None
-
     def get_queryset(self):
-        self.section_list = self.request.GET.getlist("sections")
-        self.title = self.kwargs.get("title")
-        self.part = self.kwargs.get("part")
-        query = Category.objects.filter(
-            supplementary_content__sections__title=self.title,
-            supplementary_content__sections__part=self.part,
-            supplementary_content__sections__section__in=self.section_list,
-        ).distinct()
+        section_list = self.request.GET.getlist("sections")
+        title = self.kwargs.get("title")
+        part = self.kwargs.get("part")
+        query = Category.objects.prefetch_related(
+            Prefetch(
+                'supplementary_content',
+                queryset=SupplementaryContent.objects.filter(approved=True).prefetch_related(
+                    Prefetch(
+                        'sections',
+                        queryset=RegulationSection.objects.filter(
+                            title=title,
+                            part=part,
+                            section__in=section_list,
+                        )
+                    )
+                ).distinct(),
+            )
```

Comment:
contain with something like `with_supplementary_content_for`.. its long but better than this.